loader/loader.go

```diff
@@ -787,3 +788,29 @@ func convertVolumePath(volume types.ServiceVolumeConfig) types.ServiceVolumeConf
 	volume.Source = convertedSource
 	return volume
 }
+
+func resolveEnvironment(dict map[string]any, opts *Options) {
+	for key, value := range dict {
+		if key == "environment" {
+			if environment, ok := value.([]any); ok {
+				serviceEnvironment := []any{}
+				for _, env := range environment {
+					env, ok := env.(string)
+					if !ok {
+						continue
+					}
+					if found, ok := opts.Interpolate.LookupValue(env); ok {
+						serviceEnvironment = append(serviceEnvironment, fmt.Sprintf("%s=%s", env, found))
+					} else {
+						serviceEnvironment = append(serviceEnvironment, env)
+					}
+
+				}
+				dict["environment"] = serviceEnvironment
+			}
+		}
+		if v, ok := value.(map[string]any); ok {
+			resolveEnvironment(v, opts)
+		}
+	}
```

Comment:
We don't need to loop on the whole `dict` map IHMO
```suggestion
if value, ok := dict["environment"]; ok {
		if environment, ok := value.([]any); ok {
			serviceEnvironment := []any{}
			for _, env := range environment {
				env, ok := env.(string)
				if !ok {
					continue
				}
				if found, ok := opts.Interpolate.LookupValue(env); ok {
					serviceEnvironment = append(serviceEnvironment, fmt.Sprintf("%s=%s", env, found))
				} else {
					serviceEnvironment = append(serviceEnvironment, env)
				}

			}
			dict["environment"] = serviceEnvironment
		}
		if v, ok := value.(map[string]any); ok {
			resolveEnvironment(v, opts)
		}
	}
```

Reply:
can also do 
```golang
switch environment := value.(type) {
case []any:
   for _, env := range environment {   ... }
case maps[string]any:
  resolveEnvironment(v, opts)
```